src/graph.c

```diff
@@ -113,4 +122,67 @@ uint8_t graph_cmp(const Graph* graph1, const Graph* graph2) {
   GraphSize graphSize2 = get_graph_size(graph2);
 
   return graph_size_cmp(graphSize1, graphSize2);
+}
+
+uint32_t get_max_graph_degree(const Graph* graph)
+{
+  uint32_t out_max = 0;
+  for (uint32_t v = 1; v < graph->vertices + 1; ++v)
+    if (graph->out_degrees[v] > out_max)
+      out_max = graph->out_degrees[v];
+
+  uint32_t in_max = 0;
+  for (uint32_t v = 1; v < graph->vertices + 1; ++v)
+    if (graph->in_degrees[v] > in_max)
+      in_max = graph->in_degrees[v];
+
+  return in_max > out_max ? in_max : out_max;
+}
+
+void get_graph_distribution(const Graph* graph, int32_t* distribution)
+{
```

Comment:
Znowu, jest sens dzielić obliczenia na 2 pętle? + może wygodniej by po prostu zwracać distribution z funkcji ale to tylko taka sugestia

Reply:
A i chyba by można to distribution zrobić normalnie na stosie i nie bawić się w te alokacje pamięci (?)